pkg/approver/approver_test.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package approver
+
+import (
+	"context"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	"github.com/openshift/library-go/pkg/operator/status"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	timeout = time.Second * 10
+)
+
+var _ = Describe("Cluster Operator status controller", func() {
+	var operator *configv1.ClusterOperator
+	var statusController *StatusController
+
+	BeforeEach(func() {
+		By("Running the controller")
+
+		versionGetter = status.NewVersionGetter()
+		statusController = &StatusController{
+			Client:        cl,
+			VersionGetter: versionGetter,
+		}
+		operator = &configv1.ClusterOperator{}
+		operator.SetName(clusterOperatorName)
+	})
+
+	AfterEach(func() {
+		os.Unsetenv(releaseVersionEnvVariableName)
+
+		Expect(cl.Delete(context.Background(), operator)).To(Succeed())
+		err := cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, &configv1.ClusterOperator{})
+		Expect(apierrors.IsNotFound(err)).To(BeTrue())
```

Comment:
I wonder if this could flake, maybe wrapping an eventually around this might be safer? WDYT?
```suggestion
		Eventually(func() error {
			return apierrors.IsNotFound(cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, &configv1.ClusterOperator{}))
		}, timeout).Should(BeTrue())
```